libcontainerd/remote_unix.go

```diff
@@ -414,6 +414,18 @@ func (r *remote) runContainerdDaemon() error {
 	if err := cmd.Start(); err != nil {
 		return err
 	}
+
+	// unless strictly necessary, do not add anything in between here
+	// as the reaper goroutine below needs to kick in as soon as possible
+	// and any "return" from code paths added here will defeat the reaper
+	// process.
+
+	r.daemonWaitCh = make(chan struct{})
+	go func() {
```

Comment:
Maybe we can make sure this goroutine is started before proceeding if the above note is so important (I'm not familiar with this code path).
Something like:
```go
waitCh := make(chan struct{})
go func() {
    waitCh <- struct{}{}
    cmd.Wait()
    close(waitCh)
}()

<-waitCh
r.daemonWaitCh = waitCh
```